server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -426,6 +425,11 @@ public PipelinedRequest resolvePipeline(SearchRequest searchRequest, IndexNameEx
                 pipeline = pipelineHolder.pipeline;
             }
         }
+        if (searchRequest.source() != null
+            && searchRequest.source().verbosePipeline()
+            && NOOP_PIPELINE_ID.equals(pipeline.getId()) == true) {
+            throw new IllegalArgumentException("The 'verbose pipeline' option requires a search pipeline to be defined.");
+        }
```

Comment:
Is this necessary? I guess it doesn't hurt.

Rather than looking at the `NOOP_PIPELINE_ID`, I think you can at least check for object equality against `NO_OP_PIPELINE` (since it's a singleton).